soc/xtensa/esp32/soc.h

```diff
@@ -24,9 +24,7 @@ static inline void esp32_clear_mask32(uint32_t v, uint32_t mem_addr)
 	sys_write32(sys_read32(mem_addr) & ~v, mem_addr);
 }
 
-extern int esp32_rom_intr_matrix_set(int cpu_no,
-				     int interrupt_src,
-				     int interrupt_line);
+extern void intr_matrix_set(int cpu_no, uint32_t model_num, uint32_t intr_num);
```

Comment:
Please retain the `esp32_rom_` prefix here to avoid contaminating the global namespace with symbols other modules might define.  See another comment about having two exposed symbols; Zephyr code should use the one with the prefix.